### PR TITLE
test for check_new_commits method

### PR DESCRIPTION
**Pull request by Ankitha-Sathyanarayana**
test for check_new_commits method

### CODE REVIEW

File: test_tar_changes.py [hunk 1]
```diff
@@ -75,3 +75,17 @@ def test_upload_source(delete_sources, mocked_run, mocked_sources):
     # flag was detected by argparse
     tarchanges.upload_source(osdist, 'test.tar', False)
     mocked_run.assert_called_once() # run() was not called a 2nd time
+
+@patch.object(git,'get_commit_hashes',return_value = ["9e20ef1b14ac70dea53123"])
+@patch.object(git,'get_commit_bzs')
```
Comment (bigjust) [inline test_tar_changes.py:80]: 1 space after commas in parameters

File: test_tar_changes.py [hunk 2]
```diff
@@ -75,3 +75,17 @@ def test_upload_source(delete_sources, mocked_run, mocked_sources):
...
+    # Check if empty list is being returned when old and new commits are the same
+    actual = tarchanges.check_new_commits('v12.2.8', '9e20ef1b14ac70dea53123', '9e20ef1b14ac70dea53123')
+    expected = []
+    assert actual == expected
+    # Check if get_commit_bzs is called
+    tarchanges.check_new_commits('v12.2.8', '9e20ef1b14ac70dea53123', '9e20ef1b14ac70dea53567')
+    mocked_bzs.assert_called_once()
+    # get_commit_bzs is not called
+    tarchanges.check_new_commits('v12.2.8', '8u31ef1b14ac70dea53123', '9e20ef1b14ac70dea53567')
+    mocked_bzs.assert_called_once()
```
Comment (bigjust) [inline test_tar_changes.py:91]: stylistically, i think this test case would benefit from blank lines in between assertions, so after Lines 85 and 88

**Review by bigjust — CHANGES_REQUESTED** [review]

File: test_tar_changes.py (hunk 2 above)
Comment (Ankitha-Sathyanarayana) [reply test_tar_changes.py]: Included blank lines as suggested

File: test_tar_changes.py (hunk 1 above)
Comment (Ankitha-Sathyanarayana) [reply test_tar_changes.py]: Made the suggested change

**Review by bigjust — APPROVED** [review]

File: test_tar_changes.py [hunk 3]
```diff
@@ -75,3 +75,19 @@ def test_upload_source(delete_sources, mocked_run, mocked_sources):
     # flag was detected by argparse
     tarchanges.upload_source(osdist, 'test.tar', False)
     mocked_run.assert_called_once() # run() was not called a 2nd time
+
+@patch.object(git, 'get_commit_hashes', return_value = ["9e20ef1b14ac70dea53123"])
```
Comment (bigjust) [inline test_tar_changes.py:79]: remove the spaces around the equals sign here

File: test_tar_changes.py [hunk 4]
```diff
@@ -75,3 +75,19 @@ def test_upload_source(delete_sources, mocked_run, mocked_sources):
...
+@patch.object(git, 'get_commit_bzs')
+def test_check_new_commits(mocked_bzs,mocked_git):
```
Comment (bigjust) [inline test_tar_changes.py:81]: add space after mocked_bzs

**Review by bigjust — CHANGES_REQUESTED** [review]
Looks good, just needs to be rebased, couple more minor things I noticed

File: test_tar_changes.py (hunk 3 above)
Comment (Ankitha-Sathyanarayana) [reply test_tar_changes.py]: I have made the changes

File: test_tar_changes.py (hunk 4 above)
Comment (Ankitha-Sathyanarayana) [reply test_tar_changes.py]: The changes are completed and the commits are squashed

**Review by bigjust — APPROVED** [review]

**Pull request merged**